src/shared_memory_manager.cc

```diff
@@ -264,6 +264,51 @@ OpenCudaIPCRegion(
 
   return nullptr;
 }
+
+TRITONSERVER_Error*
+GetCudaSharedMemoryRegionSize(CUdeviceptr data_ptr, size_t& shm_region_size)
+{
+  CUdeviceptr* base = nullptr;
+  CUresult result = cuMemGetAddressRange(base, &shm_region_size, data_ptr);
+  if (result != CUDA_SUCCESS) {
+    const char* errorString;
+    if (cuGetErrorString(result, &errorString) != CUDA_SUCCESS) {
+      return TRITONSERVER_ErrorNew(
+          TRITONSERVER_ERROR_INTERNAL, "Failed to get CUDA error string");
+    }
+    return TRITONSERVER_ErrorNew(
+        TRITONSERVER_ERROR_INTERNAL,
+        std::string(
+            "Failed to get CUDA address range: " + std::string(errorString))
+            .c_str());
+  }
+  return nullptr;
+}
+
+TRITONSERVER_Error*
+CheckCudaSharedMemoryRegionSize(
+    const std::string& name, CUdeviceptr data_ptr, size_t byte_size)
+{
+  size_t shm_region_size = 0;
+  auto err = GetCudaSharedMemoryRegionSize(data_ptr, shm_region_size);
+
+  // User-provided offset and byte_size should not go out-of-bounds.
+  if (err != nullptr || byte_size > shm_region_size) {
+    if (err != nullptr) {
+      // Should not pass the detailed error message back to the client.
+      LOG_ERROR << TRITONSERVER_ErrorMessage(err);
+      TRITONSERVER_ErrorDelete(err);
+    }
```

Comment:
@nnshah1 I went ahead and made some changes to this part. Let me know if this makes sense to you. I think we could remove the part where we log out `TRITONSERVER_ERROR_INTERNAL, "Failed to get shared memory information"`. I think that would be duplicated log if we already log out the detailed errors and pass the general one back to the client.